packages/@aws-cdk/aws-eks/lib/aws-auth.ts

```diff
@@ -94,6 +96,19 @@ export class AwsAuth extends Construct {
     this.accounts.push(accountId);
   }
 
+  private assertSameStack(construct: IConstruct) {
+
+    const thisStack = Stack.of(this);
+
+    if (Stack.of(construct) !== thisStack) {
+      // aws-auth is always part of the cluster stack, and since resources commonly take
+      // a dependency on the cluster, allowing those resources to be in a different stack,
+      // will create a circular dependency. granted, it won't always be the case,
+      // but we opted for the more causious and restrictive approach for now.
+      throw new Error(`${construct.node.uniqueId} must be defined in the scope of the ${thisStack.stackName} stack`);
```

Comment:
Add a reason (why not?) and also what should the user do to fix (use the construct directly?)

Reply:
What do you mean by "use the construct directly"? I don't think users are supposed to instantiate this construct themselves since it will override the `aws-auth` config map. Right? 

As far as I can tell the only thing to do is create the role/user in the cluster stack, which is already mentioned. But I can rephrase to something like:

```ts
      throw new Error(`${construct.node.uniqueId} should be defined in the scope of the ${thisStack.stackName} stack to prevent circular dependencies.`)
```

will that be reasonably clear...?

Reply:
Yes, that's good.